File: week4/task1.c
#include <stdio.h>
#include <stdbool.h>

int main() {
    printf("Integers\n");
    printf("char:        %zu byte\n", sizeof(char));
    printf("short:       %zu bytes\n", sizeof(short));
    printf("int:         %zu bytes\n", sizeof(int));
    printf("long:        %zu bytes\n", sizeof(long));
    printf("long long:   %zu bytes\n", sizeof(long long));

    printf("\nFloating-points\n");
    printf("float:       %zu bytes\n", sizeof(float));
    printf("double:      %zu bytes\n", sizeof(double));
    printf("long double: %zu bytes\n", sizeof(long double));

    printf("\nOther types\n");
    printf("bool:        %zu bytes\n", sizeof(bool));

    return 0;

}

File: week4/task1.java
package week4;

public class task1 {
    public static void main(String[] args) {
        // Divided by 8 to convert bits to bytes
        System.out.println("Integer types");
        System.out.println("byte:   " + (Byte.SIZE / 8) + " byte");
        System.out.println("short:  " + (Short.SIZE / 8) + " bytes");
        System.out.println("int:    " + (Integer.SIZE / 8) + " bytes");
        System.out.println("long:   " + (Long.SIZE / 8) + " bytes");
        System.out.println("char:   " + (Character.SIZE / 8) + " bytes");

        System.out.println("\nFloating-points");
        System.out.println("float:  " + (Float.SIZE / 8) + " bytes");
        System.out.println("double: " + (Double.SIZE / 8) + " bytes");
        System.out.println("\nOther types");
        System.out.println("boolean: (no fixed SIZE constant in Java)");
    }
}
